crates/oxc_linter/src/rules/jest/prefer_jest_mocked.rs

```diff
@@ -0,0 +1,332 @@
+    /// enforces the use of `jest.mocked()` for better type safety and readability.
+    ///
+    /// Restricted types:
+    ///
+    ///
+    /// - `jest.Mock`
+    /// - `jest.MockedFunction`
+    /// - `jest.MockedClass`
+    /// - `jest.MockedObject`
+    ///
+    /// ### Examples
+    ///
+    /// ```typescript
+    /// // invalid
+    /// (foo as jest.Mock).mockReturnValue(1);
+    /// const mock = (foo as jest.Mock).mockReturnValue(1);
+    /// (foo as unknown as jest.Mock).mockReturnValue(1);
+    /// (Obj.foo as jest.Mock).mockReturnValue(1);
+    /// ([].foo as jest.Mock).mockReturnValue(1);
+    ///
+    /// jest.mocked(foo).mockReturnValue(1);
+    /// const mock = jest.mocked(foo).mockReturnValue(1);
+    /// jest.mocked(Obj.foo).mockReturnValue(1);
+    /// jest.mocked([].foo).mockReturnValue(1);
+    /// ```
+    PreferJestMocked,
+    style,
+);
+
+impl Rule for PreferJestMocked {
+    fn run(&self, node: &AstNode, ctx: &LintContext) {
+        if let AstKind::TSAsExpression(ts_expr) = node.kind() {
+            if !matches!(ctx.nodes().parent_kind(node.id()), Some(AstKind::TSAsExpression(_))) {
+                Self::check_ts_as_expression(ts_expr, ctx);
+            }
+        } else if let AstKind::TSTypeAssertion(assert_type) = node.kind() {
+            Self::check_assert_type(assert_type, ctx);
+        }
+    }
+}
+
+const MOCK_TYPES: Set<&'static str> = phf_set! {
+    "Mock",
+    "MockedFunction",
+    "MockedClass",
+    "MockedObject",
+};
+
+impl PreferJestMocked {
+    fn check_ts_as_expression(as_expr: &TSAsExpression, ctx: &LintContext) {
+        let TSType::TSTypeReference(ts_reference) = &as_expr.type_annotation else {
+            return;
+        };
+        let arg_span = as_expr.expression.get_inner_expression().span();
+        Self::check(ts_reference, arg_span, as_expr.span, ctx);
+    }
+
+    fn check_assert_type(assert_type: &TSTypeAssertion, ctx: &LintContext) {
+        let TSType::TSTypeReference(ts_reference) = &assert_type.type_annotation else {
+            return;
+        };
+        let arg_span = assert_type.expression.get_inner_expression().span();
+        Self::check(ts_reference, arg_span, assert_type.span, ctx);
+    }
+
+    fn check(ts_reference: &TSTypeReference, arg_span: Span, span: Span, ctx: &LintContext) {
+        let TSTypeName::QualifiedName(qualified_name) = &ts_reference.type_name else {
+            return;
+        };
+        let TSTypeName::IdentifierReference(ident) = &qualified_name.left else {
+            return;
+        };
+
+        if !&ident.name.eq_ignore_ascii_case("jest")
+            || !MOCK_TYPES.contains(qualified_name.right.name.as_str())
+        {
+            return;
+        }
+
+        ctx.diagnostic_with_fix(use_jest_mocked(span), |fixer| {
+            let span_source_code = fixer.source_range(arg_span);
+            let mut formatter = fixer.codegen();
+            formatter.print_str(b"jest.mocked(");
```

Comment:
Consider using `format!` here for clarity

Reply:
Done.